packages/xo-web/src/common/sorted-table/index.js

```diff
@@ -666,10 +675,17 @@ export default class SortedTable extends Component {
                 selectedItemsIds[method](visibleItems[i].id)
               }
             })
-          : selectedItemsIds[method](item.id),
-    })
+          : selectedItemsIds[method](item.id)
+      selectedItems = filter(visibleItems, item =>
+        _selectedItemsIds.has(item.id)
+      )
+      this.setState({ selectedItemsIds: _selectedItemsIds })
+      this._previous = current
+    }
 
-    this._previous = current
+    if (onSelect !== undefined) {
+      onSelect(selectedItems)
+    }
```

Comment:
To make it simpler, I'd say you can first compute the `selectedItems` in the `if` (like you did) and *then* use it both for the `setState` and the `onSelect` call.